overlord/configstate/config/helpers.go

```diff
@@ -285,3 +286,34 @@ type ConfGetter interface {
 	Get(snapName, key string, result interface{}) error
 	GetMaybe(snapName, key string, result interface{}) error
 }
+
+// Patch sets values in cfg for the provided snap's configuration
+// based on patch.
+// patch keys can be dotted as the key argument to Set.
+// The patch is applied according to the order of its keys sorted by depth.
```

Comment:
I think it would be useful to clarify the order a bit, e.g.
```suggestion
// The patch is applied according to the order of its keys sorted by depth,
// with top-level keys applied first.
```

Reply:
thanks, I'll apply this in the follow-up.